Mathlib/Topology/Algebra/Group/Basic.lean

```diff
@@ -1726,6 +1782,57 @@ theorem exists_isCompact_isClosed_subset_isCompact_nhds_one
   rcases exists_mem_nhds_isClosed_subset L1 with ⟨K, hK, K_closed, KL⟩
   exact ⟨K, Lcomp.of_isClosed_subset K_closed KL, K_closed, KL, hK⟩
 
+/-- If a point in a topological group has a compact neighborhood, then the group is
+locally compact. -/
+@[to_additive]
+theorem IsCompact.locallyCompactSpace_of_mem_nhds_of_group {K : Set G} (hK : IsCompact K) {x : G}
+    (h : K ∈ 𝓝 x) : LocallyCompactSpace G := by
+  refine ⟨fun y n hn ↦ ?_⟩
+  have A : (y * x⁻¹) • K ∈ 𝓝 y := by
+    rw [← preimage_smul_inv]
+    exact (continuous_const_smul _).continuousAt.preimage_mem_nhds (by simpa using h)
+  rcases exists_mem_nhds_isClosed_subset (inter_mem A hn) with ⟨L, hL, L_closed, LK⟩
+  refine ⟨L, hL, LK.trans (inter_subset_right _ _), ?_⟩
+  exact (hK.smul (y * x⁻¹)).of_isClosed_subset L_closed (LK.trans (inter_subset_left _ _))
+
+/-- A topological group which is weakly locally compact is automatically locally compact. -/
+@[to_additive]
+lemma WeaklyLocallyCompactSpace.locallyCompactSpace_of_group [WeaklyLocallyCompactSpace G] :
```

Comment:
Can this be an instance?

Reply:
It creates a loop, but now loops are not supposed to be a problem. Let me try it and see if it breaks.